Add chapter about our interview process. (#4)
someengineering/company-handbook: hiring.md

> # Hiring

Reviewer: ```suggestion
# Interview Process
```
This file would also ideally be renamed.

> If you are interested in a job at Some Engineering Inc., please send your application to [hr@some.engineering](mailto:hr@some.engineering).

Reviewer: ```suggestion
If you are interested in a job at Some Engineering Inc., please send your application to [hr@some.engineering](mailto:hr@some.engineering).
```
Do we have an application, or are candidates simply submitting a resume?

Author: Simply submitting a resume and the open position is probably sufficient.

## Interview process for engineers

The interview process for engineers usually has those steps:

> - **First contact interview**
> 
>   This interview is a 30-minute video call where we get to know each other and learn about possible options and preferences.
> 
> - **Coding Challenge + Review**
> 
>   The coding challenge is a small homework task that should not take longer than 4 hours of your time.
>   You have time to prepare a pull request that we can discuss together afterward in a 20-minute review call.
>   We want to learn how you approach challenges and think a good solution should look like.
> 
> - **Technical Interview**
> 
>   In this 1-hour video call, we want to learn your technical expertise.
>   Ideally, this is a conversation about technologies, architecture, and good practice.
> 
> - **Team Interview**
> 
>   We collaborate in teams and think it is essential to feel engaged with the people you work with.
>   In this interview, you have the chance to get to know the team and vice versa.

Reviewer: ```suggestion
### Initial Contact

This interview is a 30-minute video call for us to get to know each other.

### Coding Challenge

The coding challenge is "homework" that should not take longer than 4 hours.

You will be asked to submit a pull request on GitHub with your changes, so we can discuss them during a 20-minute follow-up call.

We want to see how you think and approach challenges.

### Technical Interview

The goal of this 1-hour video call is to test your technical knowledge and expertise.

Please be prepared to discuss technologies, architecture, and best practices.

### Team Interview

We collaborate in teams and strongly believe that it is essential to feel connected and engaged with your teammates.

This interview is a chance for you to get to know the team, and vice versa.
```
Using sub-headings is best, as it allows for you to specific link to parts of the document/page.
